ironmon_tracker/data/RandomizerLog.lua

```diff
@@ -106,7 +107,7 @@ function RandomizerLog.formatInput(str)
 	str = str:gsub("’", "'")
 	str = str:gsub("é", Constants.getC("é"))
 	str = str:gsub("%[PK%]%[MN%]", "PKMN")
-	str = str:match("^%s*(.*)%s*$") or str -- remove leading/trailing spaces
+	str = str:match("^%s*(.-)%s*$") or str -- remove leading/trailing spaces
```

Comment:
the .* in the original was capturing any trailing whitespaces as it is a greedy quantifier. The %s* would always match 0 spaces. Changed it to .- to match as few characters as it could before matching any whitespaces with %s*

Reply:
My regex is not very great when it comes to understanding the need for one thing over another. As-is, the current code works great for parsing the game files and showing all of the information it needs to show to the player. Can I ask, what does this change actually do that makes a significant change? This question is partly because I'm simply not very good at regex and don't really understand this impact, and partly because I want to learn more about why you'd use `.-` over `.*`

Can you please share an example of a string that is properly formatted with your change to `.-` that is not properly formatted with the current code `.*`? (Not saying either code is better than the other, I'm just merely wanting to understand how this works and how it's beneficial)

Reply:
Absolutely!
The quantifier " * " is 'greedy', so it will match between 0 and unlimited instances of the character before, but match **as many as possible** before checking the next part of the pattern.

In the pattern "^%s\*(.\*)%s\*$" , because " . " matches any single (non-newline afaik) character, and because it will match those characters as many times as possible it will always match potential whitespace at the end of the string. The %s*$ after the capture group won't ever match anything as it has the option to match nothing and the .* before it can match the same thing.

The change I made was to use the " - " quantifier, which is the same as the " * " quantifier but will match as **few** characters as possible, so it will stop matching characters as soon as the next part of the pattern ( in this case %s*$) matches and therefore trailing whitespaces will be matched, but outside of the capture group and not contained in the returned string

Reply:
Here is some example code:

![image](https://user-images.githubusercontent.com/19582161/221722745-9a150667-31c1-4d81-92fa-a54e9952e078.png)

And the output strings for that code:

![image](https://user-images.githubusercontent.com/19582161/221722780-009e5372-a801-4060-a56d-32b5be70b77b.png)

Reply:
Thank you for the great examples. I see what you're saying here. I suppose this was just kind of working before because the outer patterns themselves were excluding the leading/trailing spaces.